proxy/http/HttpTransact.cc

```diff
@@ -520,7 +520,22 @@ HttpTransact::BadRequest(State *s)
   TxnDebug("http_trans", "[BadRequest]"
                          "parser marked request bad");
   bootstrap_state_variables_from_request(s, &s->hdr_info.client_request);
-  build_error_response(s, HTTP_STATUS_BAD_REQUEST, "Invalid HTTP Request", "request#syntax_error");
+
+  const char *body_factory_template = "request#syntax_error";
+  HTTPStatus status                 = HTTP_STATUS_BAD_REQUEST;
+  const char *reason                = "Invalid HTTP Request";
+
+  switch (s->http_return_code) {
```

Comment:
Probably better to differentiate `431 Request Header Fields Too Large` from HTTP_STATUS_REQUEST_URI_TOO_LONG

Reply:
Yea, I agree. Unfortunately, it requires to expose more finer parser result codes, which isn't 
currently possible without breaking API compatibility (`ParseResult` is defined in apidefs.h). 

Link to slack discussion with Alan on this - 
https://the-asf.slack.com/archives/CHQ1FJ9EG/p1564172180097500

Planning on revisiting this once `ATS 9` is branched out to not delay/disrupt the branching when we are so close to it.

Reply:
BTW, the current patch still preserves '400 Bad Request' for too long HTTP header fields. Only the Request URI results in 414 

https://github.com/apache/trafficserver/pull/5751/files#diff-40fc32ab3504372e0eb74a54108d1b6fR729

Reply:
👍 Thanks for the explanation.